js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorerSidebar.tsx

```diff
@@ -37,30 +48,41 @@ export const AssetGraphExplorerSidebar = React.memo(
     explorerPath,
     onChangeExplorerPath,
     allAssetKeys,
+    hideSidebar,
   }: {
     assetGraphData: GraphData;
     lastSelectedNode: GraphNode;
     selectNode: (e: React.MouseEvent<any> | React.KeyboardEvent<any>, nodeId: string) => void;
     explorerPath: ExplorerPath;
     onChangeExplorerPath: (path: ExplorerPath, mode: 'replace' | 'push') => void;
     allAssetKeys: AssetKey[];
+    hideSidebar: () => void;
   }) => {
     const selectNode: typeof _selectNode = (e, id) => {
       _selectNode(e, id);
       if (!assetGraphData.nodes[id]) {
-        const path = JSON.parse(id);
-        const nextOpsQuery = `${explorerPath.opsQuery} \"${path[path.length - 1]}\"`;
-        onChangeExplorerPath(
-          {
-            ...explorerPath,
-            opsQuery: nextOpsQuery,
-          },
-          'push',
-        );
+        try {
+          const path = JSON.parse(id);
+          const nextOpsQuery = `${explorerPath.opsQuery} \"${path[path.length - 1]}\"`;
+          onChangeExplorerPath(
+            {
+              ...explorerPath,
+              opsQuery: nextOpsQuery,
+            },
+            'push',
+          );
+        } catch (e) {
+          // Ignore errors. The selected node might be a group or code location so trying to JSON.parse the id will error.
+          // For asset nodes the id is always a JSON array
+        }
       }
     };
     const [openNodes, setOpenNodes] = React.useState<Set<string>>(new Set());
-    const [selectedNode, setSelectedNode] = React.useState<null | {id: string; path: string}>(null);
+    const [selectedNode, setSelectedNode] = React.useState<
+      null | {id: string; path: string} | {id: string}
+    >(null);
```

Comment:
Would vote for not calling this a "node" -- maybe selectedRow? The word `node` is used in a lot of the asset code to refer to an assetNode specifically so if this UI allows selecting a repository / group too maybe we change the naming

Reply:
It might also be nice to make this a union type with a dicriminator key like `type: "asset"` so people know it's "asset node with id + path or repo with id" and don't try to pass an asset node using the `{id: string}` format